app/src/main/java/org/sagebase/crf/step/CrfInstructionButtonType.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *    Copyright 2017 Sage Bionetworks
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+package org.sagebase.crf.step;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Created by TheMDP on 10/24/17.
+ */
+
+public enum  CrfInstructionButtonType {
+    @SerializedName("default")
```

Comment:
I've been experimenting with using string constants instead of enums. Here's an example:

    @Retention(SOURCE)
    @StringDef({NEXT, PREVIOUS, END})
    @interface UIActionType {
        String NEXT = "next";
        String PREVIOUS = "previous";
        String END = "end";
    }


then the strings can by "typed" 

    void handleAction(@UIActionType String actionType);

    @Override
    public void handleAction(@UIActionType String actionType) {
        TaskState taskState = new TaskState(currentStep, result);
        switch (actionType) {
            case UIActionType.NEXT:
                this.getNextStepUseCase
                        .execute(taskState, this::showNextStep, this::showStepError);
                break;
            case UIActionType.PREVIOUS:
                this.getPreviousStepUseCase
                        .execute(taskState, this::showPreviousStep, this::showStepError);
                break;
            default:
                LOGGER.warn("Unknown asyncAction type: " + actionType);
        }
    }

The main motivation is to enable the values to change over time in a non-breaking way, and I am not sure an enum allows for that (e.g. introducing a new value would break old clients that wouldn't be able to deserialize the enum). 

Just wondering if you have any thoughts about this approach vs enum

Reply:
I just tested it, and if you provide a value in json that is different than the available enums, it just doesn't set it to anything and is null, it doesn't break.

I haven't coded with StringDef before, it is new to me.  I saw it used in the CameraSource code I pulled in, where it was used to allow a sort of multiple inheritance of different String constants into a defined parameter for a method.

Reply:
Oh, that's good to know. Thank you Gson! Oh yea, that was another reason. I was annoyed how if I defined an enum in ResearchStack, I can't override/extend it because Java

Reply:
That's awesome and that always annoys me too. I wanted to do that with SurveyItemTypes. It's definitely worth looking into for backbone.  Not much help in the top-level CRF module however.